Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedMapModel.swift

```diff
@@ -64,16 +122,136 @@ class PreplannedMapModel: ObservableObject, Identifiable {
         }
     }
     
-    /// A Boolean value indicating if download can be called.
-    var canDownload: Bool {
-        switch status {
-        case .notLoaded, .loading, .loadFailure, .packaging, .packageFailure,
-                .downloading, .downloaded:
-            false
-        case .packaged, .downloadFailure:
-            true
+    /// Updates the status based on the download result of the mobile map package.
+    func updateDownloadStatus(for downloadResult: Optional<Result<MobileMapPackage, any Error>>) {
+        switch downloadResult {
+        case .success:
+            status = .downloaded
+            if let mobileMapPackage = try? downloadResult?.get() {
+                self.mobileMapPackage = mobileMapPackage
+            }
+        case .failure(let error):
+            status = .downloadFailure(error)
+        case .none:
+            if let mobileMapPackage = try? downloadResult?.get() {
+                setMobileMapPackage(mobileMapPackage)
+            }
```

Comment:
Will this branch even get visited? 🤔 `none` means result is `nil`, `get()` only works when result is `success`?

p.s., I'm still not totally convinced by the optional `result` property, but currently I cannot think of a more concise way to handle the `nil` packaging status problem. Will keep thinking 😵‍💫

Reply:
Good point, this branch is not needed.